src/test/java/com/nkh/ECommerceShop/controller/AuthControllerTest.java

```diff
@@ -0,0 +1,192 @@
+package com.nkh.ECommerceShop.controller;
+
+import com.google.gson.Gson;
+import com.nkh.ECommerceShop.dto.auth.TokenRefreshRequestDTO;
+import com.nkh.ECommerceShop.dto.auth.UserCredentialsDTO;
+import com.nkh.ECommerceShop.model.RefreshToken;
+import com.nkh.ECommerceShop.model.Role;
+import com.nkh.ECommerceShop.model.Users;
+import com.nkh.ECommerceShop.repository.RefreshTokenRepository;
+import com.nkh.ECommerceShop.repository.UsersRepository;
+import com.nkh.ECommerceShop.security.service.RefreshTokenService;
+import com.nkh.ECommerceShop.security.service.UserDetailsServiceImpl;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.ResultActions;
+
+import java.time.Instant;
+import java.util.UUID;
+
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.MOCK)
```

Comment:
No need to initilize the whole spring framework. You can use the @WebMvcTest to init just the controllers and then you can provide mocks for your repositories

Reply:
changed